src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -91,7 +91,7 @@ internal static List<SelectListItem> TriStateValues(bool? value)
 
         public static string CollectionTemplate(IHtmlHelper htmlHelper)
         {
-            var model = htmlHelper.ViewData.ModelMetadata.Model;
+            var model = htmlHelper.ViewData.Model;
```

Comment:
interesting and worth keeping an eye on.  did you happen to find any realistic scenarios where `ViewData.Model` gets out of sync with `ViewData.ModelMetadata.Model`?

suspect inconsistencies may happen in the use-`string`-instead cases.  so the old code may have been defensive.  but this template should never be used when the type is `string`.